core/src/main/kotlin/io/customer/sdk/lifecycle/CustomerIOActivityLifecycleCallbacks.kt

```diff
@@ -18,9 +19,13 @@ import kotlinx.coroutines.launch
  * and should avoid listening to events from ActivityLifecycleCallbacks directly.
  */
 class CustomerIOActivityLifecycleCallbacks : Application.ActivityLifecycleCallbacks {
-    private val lifecycleEvents = MutableSharedFlow<LifecycleStateChange>(replay = 1)
+    private val lifecycleEvents = MutableSharedFlow<LifecycleStateChange>()
     private val subscriberScope = SDKComponent.scopeProvider.lifecycleListenerScope
 
+    // Stores last emitted lifecycle state to provide current state to new subscribers.
+    // The reference is weak to avoid memory leaks.
+    private var currentLifecycleStateRef: WeakReference<LifecycleStateChange>? = null
```

Comment:
why dont we just use `state flow` if we want to have the last state stored?

Reply:
Because the problem actually lies in strong reference. Whether we use `StateFlow` or `SharedFlow` with replay, the memory leakage tests fail. My understanding is that it fails because we hold strong reference to activity in `LifecycleStateChange` class, and if for any reason, we hold  strong reference to `LifecycleStateChange`, it causes a memory leak.

So `WeakReference` here is used to have similar behavior to `StateFlow`, but with a weak reference to prevent potential memory leaks.

Reply:
That makes sense but i would make a case that we should rather keep a weak reference of `Activity` in first place? and then even if `StateFlow/Shared flow` is keeping a strong reference we would not care.

Reply:
I created another [PR](https://github.com/customerio/customerio-android/pull/385) with the strategy mentioned above, if it sounds good lets merge that otherwise, we can keep it otherwise we can go with this one.

Reply:
Thanks Shahroz for creating the PR, added some comments on it.